test/Main.hs

```diff
@@ -1257,6 +1257,49 @@ testSqlWith step = do
       (results :: [T.Text]) <- fetchMany runIdentity
       liftIO $ assertEqual "Wrong number of banks left" 2 (length results)
 
+testUnion :: HasCallStack => (String -> TestM ()) -> TestM ()
+testUnion step = do
+  step "Running SQL UNION tests"
+  testPass
+  where
+    testPass = do
+      runQuery_ . sqlSelect "" $ do
+        sqlResult "true"
+        sqlUnion
+          [ sqlSelect "" $ do
+              sqlResult "false"
+              sqlUnion
+                [ sqlSelect "" $ do
+                    sqlResult "true"
+                ]
+          ]
+      result <- fetchMany runIdentity
+      liftIO $ assertEqual "UNION of booleans"
+        [False, True]
+        result
+
+
+testUnionAll :: HasCallStack => (String -> TestM ()) -> TestM ()
+testUnionAll step = do
+  step "Running SQL UNION ALL tests"
+  testPass
+  where
+    testPass = do
+      runQuery_ . sqlSelect "" $ do
+        sqlResult "true"
+        sqlUnionAll
+          [ sqlSelect "" $ do
+              sqlResult "false"
+              sqlUnionAll
```

Comment:
I'm not too chuffed by how this has to be nested inside the first UNION instead of being at the same level, but I'm not sure if it's desirable in the first place.